src/server/src/main/java/io/cassandrareaper/service/RepairManager.java

```diff
@@ -83,13 +87,21 @@ private RepairManager(
 
   public static RepairManager create(
       AppContext context,
+      ClusterFacade clusterFacade,
```

Comment:
remove this.

Instead, add a @VisibleForTesting package-protected static method like…

```
  private RepairManager(
      AppContext context,
      Supplier<ClusterFacade> clusterFacadeSupplier,
      ScheduledExecutorService executor,
      long repairTimeout,
      TimeUnit repairTimeoutTimeUnit,
      long retryDelay,
      TimeUnit retryDelayTimeUnit)  {

     …
  }

  @VisibleForTesting
  static RepairManager create(
      AppContext context,
      Supplier<ClusterFacade> clusterFacadeSupplier,
      ScheduledExecutorService executor,
      long repairTimeout,
      TimeUnit repairTimeoutTimeUnit,
      long retryDelay,
      TimeUnit retryDelayTimeUnit) {

    return new RepairManager(context, clusterFacadeSupplier, executor, repairTimeout, repairTimeoutTimeUnit, retryDelay, retryDelayTimeUnit);
  }

  // the default factory method delegates to the default facade behaviour
  public static RepairManager create(
      AppContext context,
      ScheduledExecutorService executor,
      long repairTimeout,
      TimeUnit repairTimeoutTimeUnit,
      long retryDelay,
      TimeUnit retryDelayTimeUnit) {

    return create(context, () -> ClusterFacade.create(context), executor, repairTimeout, repairTimeoutTimeUnit, retryDelay, retryDelayTimeUnit);
}

That allows you to pass in the mocks for testing, without (overly) polluting production code.

And if you find yourself having to inject ClusterFacade mocks for tests outside the package, it might be raising the question of why you require mocking something that's two levels away from the current layer you are *unit* testing.

Reply:
done